lib/plugins/aws/custom-resources/index.js

```diff
@@ -179,6 +179,19 @@ async function addCustomResourceToService(awsProvider, resourceName, iamRoleStat
   };
   Resources[customResourceFunctionLogicalId] = customResourceFunction;
 
+  // configure vpc for the custom resource function
+  const vpcConfig = providerConfig.vpc || { securityGroupIds: null, subnetIds: null };
+  customResourceFunction.Properties.VpcConfig = {
+    SecurityGroupIds: vpcConfig.securityGroupIds,
+    SubnetIds: vpcConfig.subnetIds,
+  };
+  if (
+    !customResourceFunction.Properties.VpcConfig.SecurityGroupIds ||
+    !customResourceFunction.Properties.VpcConfig.SubnetIds
+  ) {
+    delete customResourceFunction.Properties.VpcConfig;
+  }
```

Comment:
Why simply not:

```javascript
if (providerConfig.vpc) {
  customResourceFunction.Properties.VpcConfig = providerConfig.vpc
}
```

?

Reply:
I thought of covering the case where user might not provide subnetIds or securityGroupIds.

Reply:
Ok, in such case we still can write it in similar way:

```javascript
if (providerConfig.vpc && provider.vpc.securityGroupIds && provider.vpc.subnetIds) {
  customResourceFunction.Properties.VpcConfig = providerConfig.vpc
}
```